### PR TITLE
backend: add db transactions and network calls principle

### PR DESCRIPTION
**Pull request by divineforest**

### CODE REVIEW

File: src/backend.md [hunk 1]
```diff
@@ -169,6 +169,28 @@ Method 3 is the same except you implement your migration in lambda function and
 
 For serverless projects - method 3 is preferred, but not always. When you have a huge dataset and it is impossible to perform migration within one serverless function invocation because of timeout (15 minutes usually), method 2 is the only option since it allows you to split your migration by batches via multiple SQS messages.
 
+### Transactions
+
+1. Avoid initiating read network calls within a transaction before the first SQL statement.
```
Comment (andrey-kuznetsov-osome) [inline src/backend.md:174]: Hmm.

Why it is only about **before SQL**? I think all network operation should be placed before or into `afterCommit` statement or better to separate jobs via transaction outbox.

Comment (andrey-kuznetsov-osome) [reply src/backend.md:174]: Imagine you place you async network call like:
```ts
return sequelize.transaction(async (transaction) => {
     await User.save({ name }, { where: { id: cachedId } });
     // long operation
     const cachedId = await redis.get('userId');
   });
```

You also prevents commit of this transaction with additional wait

Comment (dmlyapin) [reply src/backend.md:174]: @andrey-kuznetsov-osome 
At least, we can move all network calls that placed before the first db call without doubts. Moving network calls from the middle could be more tricky.

**Review by dmlyapin — APPROVED** [review]

File: src/backend.md (hunk 1 above)
Comment (divineforest) [reply src/backend.md:174]: yes, I used same logic. About the "after" it's more related to atomic updates and is a separate principle. Will do next

**Pull request merged**